src/main/java/org/folio/rest/service/LogContextService.java

```diff
@@ -1,87 +1,40 @@
 import org.folio.rest.repository.FeeFineRepository;
-import org.joda.time.DateTime;
 
 import java.util.Map;
 
 public class LogContextService {
-  public static final String STAFF_INFO_ONLY = "Staff info only";
-  public static final String BILLED = "Billed";
-
   private final FeeFineRepository feeFineRepository;
   private final AccountRepository accountRepository;
   private final UsersClient usersClient;
-  private FeeFineLogContext context;
 
   public LogContextService(Vertx vertx, Map<String, String> okapiHeaders) {
     PostgresClient pgClient = PgUtil.postgresClient(vertx.getOrCreateContext(), okapiHeaders);
-
     feeFineRepository = new FeeFineRepository(pgClient);
     accountRepository = new AccountRepository(pgClient);
-
     usersClient = new UsersClient(vertx, okapiHeaders);
   }
 
-  public Future<FeeFineLogContext> createLogContext(Feefineaction action) {
-    context = new FeeFineLogContext()
-      .withDate(DateTime.now())
-      .withType(action.getTypeAction())
-      .withAmount(action.getAmountAction())
-      .withSource(action.getSource())
-      .withServicePointId(action.getCreatedAt());
-
-    if (isAction(action)) {
-      context = context.withAction(action.getTypeAction())
-        .withBalance(action.getBalance())
-        .withPaymentMethod(action.getPaymentMethod())
-        .withComment(action.getComments());
-    } else if (STAFF_INFO_ONLY.equalsIgnoreCase(action.getTypeAction())) {
-      context = context.withAction(STAFF_INFO_ONLY);
-    } else if (isCharge(action)) {
-      context = context.withAction(BILLED);
-    }
-
+  public Future<JsonObject> createFeeFineLogContext(Feefineaction action) {
     return accountRepository.getAccountById(action.getAccountId())
-      .compose(this::processAccount);
-  }
-
-  private Future<FeeFineLogContext> processAccount(Account account) {
-    context = context.withUserId(account.getUserId())
-      .withFeeFineId(account.getFeeFineId())
-      .withFeeFineOwner(account.getFeeFineOwner())
-      .withItemBarcode(account.getBarcode())
-      .withLoanId(account.getLoanId());
-
-    return usersClient.fetchUserById(account.getUserId())
-      .compose(this::addUserData)
-      .compose(c -> feeFineRepository.getById(account.getFeeFineId()))
-      .compose(this::addFeeFineData);
-  }
-
-  private Future<FeeFineLogContext> addUserData(User user) {
-    context = context.withUserBarcode(user.getBarcode());
-    return succeededFuture(context);
+      .compose(account -> createFeeFineLogContext(action, account));
   }
 
-  private Future<FeeFineLogContext> addFeeFineData(Feefine feefine) {
-    ofNullable(feefine).ifPresent(f -> context = context.withAutomated(f.getAutomatic()));
-    return succeededFuture(context);
+  public Future<JsonObject> createFeeFineLogContext(Feefineaction action, Account account) {
+    return feeFineRepository.getById(account.getFeeFineId())
+      .compose(feeFine -> usersClient.fetchUserById(account.getUserId())
```

Comment:
What is a reason to fetch user by id here and introduce additional call here? As I remember we discussed fetching user in the mod-audit to don't affect other business flows.